main.nf

```diff
@@ -421,26 +421,27 @@ process samtools {
 
 
 /*
- * STEP 3.3 - Statistics about unmapped reads against ref genome
+ * STEP 3.3 - Statistics about mapped reads against ref genome
  */
 
-process bwa_unmapped {
+process bwa_mapped {
     tag "${input_files[0].baseName}"
-    publishDir "${params.outdir}/bwa/unmapped", mode: 'copy'
+    publishDir "${params.outdir}/bwa/mapped", mode: 'copy'
 
     input:
-    file input_files from bam_for_unmapped.collect()
+    file input_files from bam_for_mapped.collect()
+    file bai from bai_for_mapped.collect()
 
     output:
-    file 'unmapped_refgenome.txt' into bwa_unmapped
+    file 'mapped_refgenome.txt' into bwa_mapped
 
     script:
     """
     for i in $input_files
     do
       printf "\${i}\t"
-      samtools view -c -f0x4 \${i}
-    done > unmapped_refgenome.txt
+      samtools idxstats \${i} | awk '{s+=\$3}END{print s}'
+    done > mapped_refgenome.txt
```

Comment:
Ah, I guess it's to get a single merged file. Fair enough.. I still think that the performance would be better with parallel samtools processes which are then merged afterwards in a separate process, but I guess it's not a very time consuming step so it's probably fine as it is..

Reply:
Yes, using the idxstats methods it can be completed in seconds.